lib/ansible/plugins/lookup/aws_sg_transform.py

```diff
@@ -0,0 +1,112 @@
+# (c) 2017, Aaron Haaf <aabonh(at)gmail.com>
+# (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.ec2 import HAS_BOTO3
+from ansible.module_utils.ec2 import get_aws_connection_info
+from ansible.module_utils.ec2 import get_ec2_names_from_security_group_ids
+from ansible.module_utils.ec2 import get_ec2_security_group_ids_from_names
+from ansible.module_utils.parsing.convert_bool import boolean
+from ansible.plugins.lookup import LookupBase
+
+try:
+    import boto3
+except ImportError:
+    pass  # will be captured by imported HAS_BOTO3
+
+DOCUMENTATION = '''
+lookup: aws_ssm
+author: Aaron Haaf <aabonh(at)gmail.com>
+version_added: 2.5
+short_description: Convert between name and id for AWS EC2 Security Groups
+description: Changes names like "My-Group-A" to an identifier like sg-2hg21da
+options:
+    aws_profile:
+    description: The boto profile to use. You may use environment variables or the default profile as an alternative.
+    region:
+    description: The region to use. You may use environment variables or the default profile's region as an alternative.
+    id_to_name:
+    description: Go backwards and convert identifiers to names
+    default: False
+'''
+
+EXAMPLES = '''
+# lookup sample:
+- name: Lookup a single group
+  debug: msg="{{lookup('aws_sg_transform', "MyGroupA")}}"
+  #=> ['sg-A']
+
+- name: Lookup a few groups in a different region
+  debug: msg="{{lookup('aws_sg_transform', "MyGroupA", "MyGroupB", region="eu-west-1")}}"
+  #=> [ ['sg-A'], ['sg-B'] ]
+
+# Multiple groups:
+- name: Get a few groups going
+  set_fact:
+    my_groups_a: ['MyGroupA', 'MyGroupB'] #=> ['sg-A', 'sg-B']
+    my_groups_b: ['MyGroupC', 'MyGroupD'] #=> ['sg-C', 'sg-D']
+
+- name: Look up a few groups in a different region
+  debug: msg="{{lookup('aws_sg_transform', my_groups_a, 'MyGroupC', region="eu-west-1")}}"
+  #=> [ ['sg-A', 'sg-B'], ['sg-C'] ]
+
+- name: Go backwards you crazy diamond
+  debug: msg="{{ lookup('aws_sg_transform', 'sg-A', id_to_name=True)}}"
+  #=> ['MyGroupA']
+'''
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables, **kwargs):
+        '''
+            :param terms: a list of plugin options
+                          e.g. [ [sg-A, sg-B], sg-C ]
+            :param variables: variables for the current host
+            :return An array of strings
+        '''
+
+        if not HAS_BOTO3:
+            raise AnsibleError('botocore and boto3 are required.')
+
+        fauxdule = self.ModuleFacade(kwargs)
```

Comment:
I am personally not a fan of faking the Module, but I couldn't find a different way to hook in and get what aws "is" like you would in a module. The goal was to allow for ease of use and the path of least surprise.

Reply:
My recommendation, would be to abstract some of the code, by removing it from `get_aws_connection_info` and into another helper.  Then `get_aws_connection_info` would accept a module, parse out what it needs, and pass it to the other helper.

Then your lookup uses that helper instead of the function that expects a module object.

Reply:
So refactoring `get_aws_connection_info` into something more generic, and making say `get_aws_connection_info_for_module`.

Reply:
Yes, look at my update to the last comment where I provided an example diff via gist.  It moves all of the real logic into a function that accepts all the necessary args, and `get_aws_connection_info` just calls it, passing in info from `module`

A few other things need handled, like exceptions so that `get_aws_connection_info` can use `module.fail_json`.

Just an example, and completely not tested.

Reply:
I'd probably leave `get_aws_connection_info` as is for backwards compatibility.  The new function would be the one you would use.